dietpi/dietpi-software

```diff
@@ -8385,6 +8416,195 @@ _EOF_
+						orport=$G_WHIP_RETURNED_VALUE
+						break
+					else
+						invalid_text='[ERROR] No valid entry found, value needs to be a sequence of integers. Please retry...\n\n'
+					fi
+				done
+
+				# Grant capability to bind to privileged ports below 1024
+				(( $obfs4port < 1024 )) && G_EXEC setcap cap_net_bind_service=+ep "$(command -v obfs4proxy)"
+
+				cat << _EOF_ > /etc/tor/torrc
+BridgeRelay 1
+ORPort $orport
+ServerTransportPlugin obfs4 exec $(command -v obfs4proxy)
+ServerTransportListenAddr obfs4 0.0.0.0:$obfs4port
+ExtORPort auto
+_EOF_
+				G_EXEC mkdir -p /etc/systemd/system/tor@.service.d
+				G_EXEC eval "echo -e '[Service]\nNoNewPrivileges=no' > /etc/systemd/system/tor@.service.d/dietpi-relay.conf"
+				G_EXEC systemctl daemon-reload
+		
+			elif [[ $G_WHIP_RETURNED_VALUE == 'Guard/Middle' ]]
+			then
+		
+				cat << _EOF_ > /etc/tor/torrc
+ORPort $orport
+ExitRelay 0
+SocksPort 0
+ControlSocket 0
+_EOF_
+			elif [[ $G_WHIP_RETURNED_VALUE == 'Exit' ]]
+			then
+				G_AGI unbound
+				aSOFTWARE_INSTALL_STATE[182]=1
+
+				cat << _EOF_ > /etc/tor/torrc
+ORPort $orport
+ExitRelay 1
+SocksPort 0
+ControlSocket 0
+
+ExitPolicy accept *:20-21
+ExitPolicy accept *:43
+ExitPolicy accept *:53
+ExitPolicy accept *:79
+ExitPolicy accept *:80-81
+ExitPolicy accept *:88
+ExitPolicy accept *:110
+ExitPolicy accept *:143
+ExitPolicy accept *:220
+ExitPolicy accept *:389
+ExitPolicy accept *:443
+ExitPolicy accept *:464
+ExitPolicy accept *:531
+ExitPolicy accept *:543-544
+ExitPolicy accept *:554
+ExitPolicy accept *:636
+ExitPolicy accept *:706
+ExitPolicy accept *:749
+ExitPolicy accept *:873
+ExitPolicy accept *:902-904
+ExitPolicy accept *:981
+ExitPolicy accept *:989-990
+ExitPolicy accept *:991
+ExitPolicy accept *:992
+ExitPolicy accept *:993
+ExitPolicy accept *:995
+ExitPolicy accept *:1194
+ExitPolicy accept *:1220
+ExitPolicy accept *:1293
+ExitPolicy accept *:1500
+ExitPolicy accept *:1533
+ExitPolicy accept *:1677
+ExitPolicy accept *:1723
+ExitPolicy accept *:1755
+ExitPolicy accept *:1863
+ExitPolicy accept *:2082
+ExitPolicy accept *:2083
+ExitPolicy accept *:2086-2087
+ExitPolicy accept *:2095-2096
+ExitPolicy accept *:2102-2104
+ExitPolicy accept *:3690
+ExitPolicy accept *:4321
+ExitPolicy accept *:4643
+ExitPolicy accept *:5050
+ExitPolicy accept *:5190
+ExitPolicy accept *:5222-5223
+ExitPolicy accept *:5228
+ExitPolicy accept *:8008
+ExitPolicy accept *:8074
+ExitPolicy accept *:8082
+ExitPolicy accept *:8087-8088
+ExitPolicy accept *:8232-8233
+ExitPolicy accept *:8332-8333
+ExitPolicy accept *:8443
+ExitPolicy accept *:8888
+ExitPolicy accept *:9418
+ExitPolicy accept *:10000
+ExitPolicy accept *:11371
+ExitPolicy accept *:19294
+ExitPolicy accept *:19638
+ExitPolicy accept *:50002
+ExitPolicy accept *:64738
+ExitPolicy reject *:*
+_EOF_
+				G_CONFIG_INJECT 'nameserver[[:blank:]]' 'nameserver 127.0.0.1' /etc/resolv.conf
```

Comment:
This may fail if there are multiple nameserver entries (`G_CONFIG_INJECT` requires the same setting to be present at most once) and it wont be persistent if `resolvconf` is installed, in which case `/etc/network/interfaces` needs to be altered to apply a persistent nameserver.

Currently there is no CLI to apply a DNS nameserver via DietPi tools,, but I'm working on one. Until then:
```
command -v resolvconf > /dev/null && G_EXEC sed -i 's/dns-nameservers.*$/dns-nameservers 127.0.0.1/' /etc/network/interfaces
G_EXEC eval "echo 'nameserver 127.0.0.1' > /etc/resolv.conf" # Unbound is running and configured in the very next config step, so it is safe to switch the DNS nameserver right now.
```